azurerm/internal/services/monitor/resource_arm_monitor_action_rule.go

```diff
@@ -0,0 +1,591 @@
+									},
+
+									"recurrence_weekly": {
+										Type:          schema.TypeSet,
+										Optional:      true,
+										MinItems:      1,
+										ConflictsWith: []string{"suppression.0.schedule.0.recurrence_monthly"},
+										Elem: &schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validation.StringInSlice(weekDays, false),
+										},
+									},
+
+									"recurrence_monthly": {
+										Type:          schema.TypeSet,
+										Optional:      true,
+										ConflictsWith: []string{"suppression.0.schedule.0.recurrence_weekly"},
+										Elem: &schema.Schema{
+											Type:         schema.TypeInt,
+											ValidateFunc: validation.IntBetween(1, 31),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"condition": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alert_context": schemaActionRuleAlertContextCondtion(),
+
+						"alert_rule_id": schemaActionRuleAlertRuleIDCondtion(),
+
+						"description": schemaActionRuleDescriptionCondtion(),
+
+						"monitor": schemaActionRuleMonitorCondtion(),
+
+						"monitor_service": schemaActionRuleMonitorServiceCondtion(),
+
+						"severity": schemaActionRuleSeverityCondtion(),
+
+						"target_resource_type": schemaActionRuleTargetResourceTypeCondtion(),
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmMonitorActionRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Monitor.ActionRulesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.GetByName(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing Monitor ActionRule %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_monitor_action_rule", *existing.ID)
+		}
+	}
+
+	actionRule := alertsmanagement.ActionRule{
+		// the location is always global from the portal
+		Location: utils.String(location.Normalize("Global")),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	t := alertsmanagement.Type(d.Get("type").(string))
+	description := d.Get("description").(string)
+	scope := expandArmActionRuleScope(d.Get("scope").([]interface{}))
+	conditions := expandArmActionRuleConditions(d.Get("condition").([]interface{}))
+
+	actionRuleStatus := alertsmanagement.Enabled
+	if !d.Get("enabled").(bool) {
+		actionRuleStatus = alertsmanagement.Disabled
+	}
+
+	switch t {
+	case alertsmanagement.TypeSuppression:
```

Comment:
Given that these are adding additional complexity - it seems like it might make sense to split out this resource into three different ones? one for suppression, action group and diagnostic?